internal/service/pushbasedlogexport/resource_test.go

```diff
@@ -0,0 +1,311 @@
+				Check:  resource.ComposeTestCheckFunc(commonChecks(s3BucketName1, nonEmptyPrefixPath)...),
+			},
+			{
+				Config: configBasicUpdated(projectName, orgID, s3BucketName1, s3BucketName2, s3BucketPolicyName, awsIAMRoleName, awsIAMRolePolicyName, nonEmptyPrefixPath, true),
+				Check:  resource.ComposeTestCheckFunc(commonChecks(s3BucketName2, nonEmptyPrefixPath)...),
+			},
+			{
+				Config:                               configBasicUpdated(projectName, orgID, s3BucketName1, s3BucketName2, s3BucketPolicyName, awsIAMRoleName, awsIAMRolePolicyName, nonEmptyPrefixPath, true),
+				ResourceName:                         resourceName,
+				ImportStateIdFunc:                    importStateIDFunc(resourceName),
+				ImportState:                          true,
+				ImportStateVerify:                    true,
+				ImportStateVerifyIdentifierAttribute: "project_id",
+			},
+		},
+	})
+}
+
+func TestAccPushBasedLogExport_noPrefixPath(t *testing.T) {
+	var (
+		orgID                = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName          = acc.RandomProjectName()
+		s3BucketNamePrefix   = fmt.Sprintf("tf-%s", acc.RandomName())
+		s3BucketName1        = fmt.Sprintf("%s-1", s3BucketNamePrefix)
+		s3BucketName2        = fmt.Sprintf("%s-2", s3BucketNamePrefix)
+		s3BucketPolicyName   = fmt.Sprintf("%s-s3-policy", s3BucketNamePrefix)
+		awsIAMRoleName       = acc.RandomIAMRole()
+		awsIAMRolePolicyName = fmt.Sprintf("%s-policy", awsIAMRoleName)
+	)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ExternalProviders:        acc.ExternalProvidersOnlyAWS(),
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: configBasic(projectName, orgID, s3BucketName1, s3BucketName2, s3BucketPolicyName, awsIAMRoleName, awsIAMRolePolicyName, defaultPrefixPath, false),
+				Check:  resource.ComposeTestCheckFunc(commonChecks(s3BucketName1, defaultPrefixPath)...),
+			},
+		},
+	})
+}
+
+func commonChecks(s3BucketName, prefixPath string) []resource.TestCheckFunc {
+	attributes := map[string]string{
+		"bucket_name": s3BucketName,
+		"prefix_path": prefixPath,
+	}
+	checks := addAttrChecks(nil, attributes)
+	checks = acc.AddAttrSetChecks(resourceName, checks, "project_id", "iam_role_id")
+	return acc.AddAttrSetChecks(datasourceName, checks, "project_id", "iam_role_id")
+}
+
+func addAttrChecks(checks []resource.TestCheckFunc, mapChecks map[string]string) []resource.TestCheckFunc {
+	checks = acc.AddAttrChecks(resourceName, checks, mapChecks)
+	return acc.AddAttrChecks(datasourceName, checks, mapChecks)
+}
+
+func configBasic(projectName, orgID, s3BucketName1, s3BucketName2, s3BucketPolicyName, awsIAMRoleName, awsIAMRolePolicyName, prefixPath string, usePrefixPath bool) string {
+	test := fmt.Sprintf(`
+	 	locals {
+		 		project_name = %[1]q
```

Comment:
as comment above, evaluate if ProjectIDExecution can be used so project_id is passed instead of projectName and orgID

Reply:
This resource is configured per project so we can't use ProjectIDExecution

Reply:
updated to use ProjectIDExecution